.github/workflows/functional-test-noncloud.yaml

```diff
@@ -0,0 +1,708 @@
+# ------------------------------------------------------------
+# Copyright 2023 The Radius Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#    
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------
+
+name: Functional tests with no cloud resources
+
```

Comment:
Can we consolidate this with some of the other workflows? (why or why not?). Without adding more complexity we're going to have at least 3 flavors:

- Scheduled Azure tests
- PR-based non-cloud functional tests
- PR-based functional tests

I'm just thinking about how we can make this more maintainable. Do you have ideas?